### ISSUE TITLE
sql "is [not] distinct from" expressions are incorrectly split by "from"

### ISSUE THREAD

**Issue by IgnorantWalking** [tconbeer/sqlfmt#327]
**Describe the bug**
When formatting a _"is [not] distinct from"_ SQL expression ([postgreSQL](https://wiki.postgresql.org/wiki/Is_distinct_from) [snowflake](https://docs.snowflake.com/en/sql-reference/functions/is-distinct-from.html)), the library seems to look for the _"from"_ part and uses it to split the expression into multiple lines, even when exists enough horizontal spaces to keep the expression untouched.

**To Reproduce**
Example input expression:
```sql
select foo, bar, baz
from dual
where foo is not distinct from baz or bar is distinct from baz
```

**Expected behavior**
The _"is [not] distinct from"_ expressions should be kept in a single line:

```sql
select foo, bar, baz
from dual
where foo is not distinct from baz 
or bar is distinct from baz
```

**Actual behavior**
```sql
select foo, bar, baz
from dual
where foo is not distinct
from baz or bar is distinct
from baz
```

**Additional context**
tested with sqlfmt v0.13.0

Many, many thanks for the library!! Even in the current early stage is a really useful tool for us!

**Comment by tconbeer** [comment]
Thanks for the report. Never heard of "is distinct from" before. Will be an easy add.

**Comment by kadekillary** [comment]
@tconbeer If you're open to it - I'm happy to work on implementing this. Seems like a simple add to [Rule(name="word_operator", ...)](https://github.com/tconbeer/sqlfmt/blob/bd08b425dc3357f2480adec0639a5f9adc8df9e6/src/sqlfmt/rules/__init__.py#L43) in `sqlfmt/rules/__init__.py`.

Let me know!

**Comment by tconbeer** [comment]
Yes, exactly! And add tests, please!

**Comment by kadekillary** [comment]
@tconbeer lost on step 3 from [here](https://github.com/tconbeer/sqlfmt#updating-primer-repos-to-reflect-formatting-changes) - don't see any unformatted tag, thus can't checkout branch. What am I missing?

**Comment by tconbeer** [comment]
Don't worry about updating the primer -- you probably don't need to for this change. Please just open a PR and it'll run in CI. I'll help you with primer changes if necessary

**Comment by tconbeer** [comment]
closed by #332
